src/traits/marker/hand_gun.rs

```diff
@@ -39,34 +56,71 @@ fn remove_hand_gun(entity: &mut EntityCommands, slot: SlotKey) -> Result<(), Err
 	Ok(())
 }
 
-impl GetMarkerMeta for HandGun {
-	fn marker() -> MarkerMeta {
-		MarkerMeta {
-			insert_fn: add_hand_gun,
-			remove_fn: remove_hand_gun,
-		}
+fn insert_dual_fn(entity: &mut EntityCommands, slot: SlotKey) -> Result<(), Error> {
+	let SlotKey::Hand(side) = slot else {
+		return Err(slot_error(slot));
+	};
+
+	match side {
+		Side::Right => entity.insert(Marker::<(HandGun, Right, Dual)>::new()),
+		Side::Left => entity.insert(Marker::<(HandGun, Left, Dual)>::new()),
+	};
+
+	Ok(())
+}
+
+fn remove_dual_fn(entity: &mut EntityCommands, slot: SlotKey) -> Result<(), Error> {
+	let SlotKey::Hand(side) = slot else {
+		return Err(slot_error(slot));
+	};
+
+	match side {
+		Side::Right => entity.remove::<Marker<(HandGun, Right, Dual)>>(),
+		Side::Left => entity.remove::<Marker<(HandGun, Left, Dual)>>(),
+	};
+
+	Ok(())
+}
+
+fn soft_override(running: &mut Skill<Active>, new: &mut Skill<Queued>) -> bool {
+	if running.marker.tag != Some(Tag::HandGun) {
+		running.data.ignore_after_cast = false;
+		return false;
+	}
+	if new.marker.tag != Some(Tag::HandGun) {
+		running.data.ignore_after_cast = false;
+		return false;
+	}
+	if running.data.slot != new.data.slot {
```

Comment:
Matching slots should set ignore to false to avoid aftercast circumvention on quick q -> e -> q inqut.

Reply:
done